ui/components/ui/identicon/identicon.component.js

```diff
@@ -84,13 +85,11 @@ export default class Identicon extends PureComponent {
       return this.renderImage();
     }
 
-    if (address) {
-      if (isHex(address)) {
-        const checksummedAddress = checksumAddress(address);
+    if (isValidHexAddress(address)) {
```

Comment:
Also, this might prevent the display of identicons for any invalid addresses, instead showing a plain circle. We use identicons in _a lot_ of places, so I would expect that some of them will still have "invalid" addresses.

Reply:
wait, so what are the cases we want to not show a jazzicon right now? that might be easier to reason about? IF jazzicons allow non hex strings should we just move the part where we render the blockie or jazzicon (if there isn't a matching entry in the contractMap) out one level and remove the fallback render?